### PR TITLE
MM-44718 finish run extacted to hook 

### PR DESCRIPTION
**Pull request by trilopin**
#### Summary
Extract finish run behavior to a hook to share among context menu and finish run section.

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-44718

#### Checklist
- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [ ] ~~Unit tests updated~~

### CODE REVIEW

File: webapp/src/components/backstage/playbook_runs/playbook_run/hoc_finish_run.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import {finishRun} from 'src/client';
+import {modals} from 'src/webapp_globals';
+import {makeUncontrolledConfirmModalDefinition} from 'src/components/widgets/confirmation_modal';
+
+interface WithFinishRun {
+    playbookRun: PlaybookRun;
+    children: React.ReactNode;
+}
+
+export default function withFinishRun<T extends WithFinishRun>(ChildComponent: React.ComponentType<T & {onClick: any}>) {
```
Comment (trilopin) [inline webapp/src/components/backstage/playbook_runs/playbook_run/hoc_finish_run.tsx:16]: not happy with that `any` but I spend too much time already trying to make it perfect.

Suggestions & TS wisdom are more than welcome

**Comment by trilopin** [comment]
3/5 on this approach, if you feel this is too clever I'd fine changing it to a component composition approach.

**Review by agarciamontoro — COMMENTED** [review]
This is really cool! But, as you said, I'm concerned this is too clever, it took me a while to understand how everything fits together. Another solution is to have a custom hook, something similar to this:
```js
const useOnFinishRunClick = (playbookRun: PlaybookRun) => {
    const dispatch = useDispatch();
    const {formatMessage} = useIntl();
    
    const onFinishRunClick = () => {
        // ... the whole onFinishRunClick logic
    }
    
    return onFinishRunClick;
}
```
Then you could call it in the components you need as in
```
const onFinishRunClick = useOnFinishRunClick(props.playbookRun)
```

Thoughts?

**Review by agarciamontoro — APPROVED** [review]
Yay! This is easier to understand for my old tired brain. Thanks :tada:

**Pull request merged**